TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobianSSE.icc

```diff
@@ -1,9 +1,9 @@
 
-void AnalyticalCurvilinearJacobian::computeFullJacobian(const GlobalTrajectoryParameters& globalParameters,
-                                                        const GlobalPoint& x,
-                                                        const GlobalVector& p,
-                                                        const GlobalVector& h,
-                                                        const double& s) {
+inline void AnalyticalCurvilinearJacobian::computeFullJacobian(const GlobalTrajectoryParameters& globalParameters,
```

Comment:
@makortel , looks like this change caused the build error https://github.com/cms-sw/cmssw/pull/34140#issuecomment-861650287

Reply:
I woiuld suggest to add //NOLINT(misc-definitions-in-headers) for this

Reply:
Hmm. In this case the `AnalyticalCurvilinearJacobianSSE.icc` (and `AnalyticalCurvilinearJacobianEXT.icc`) are more source-like than header-like files. An easy way would be to rename these files to use another file extension, but that was disfavored earlier (https://github.com/cms-sw/cmssw/pull/33366#discussion_r611747942).

An alternative would be to rename those files to `.cc`, and move the `#ifdef`s there along
```cpp
// in AnalyticalCurvilinearJacobian.cc
#if !(defined(USE_SSEVECT) || defined(USE_EXTVECT)) || defined(TRPRFN_SCALAR)
void AnalyticalCurvilinearJacobian::computeFullJacobian(...) { ... }
#endif

// in AnalyticalCurvilinearJacobianSSE.cc
#if defined(USE_SSEVECT) && !defined(TRPRFN_SCALAR)
void AnalyticalCurvilinearJacobian::computeFullJacobian(...) { ... }
#endif

// in AnalyticalCurvilinearJacobianEXT.cc
#if defined(USE_EXTVECT) && !defined(TRPRFN_SCALAR)
void AnalyticalCurvilinearJacobian::computeFullJacobian(...) { ... }
#endif
```
but keeping the `#else` case manually up to date would be prone to errors.

So maybe the `NOLINT` would be least bad.

Reply:
thanks @makortel , I have added `// NOLINTNEXTLINE(misc-definitions-in-headers)` for this